### ISSUE TITLE
Mass assignment false positive on merging AR object into strong parameters

### ISSUE THREAD

**Issue by JasonBarnabe** [presidentbeef/brakeman#1054]
I get many false positives in my project with this kind of code:

```ruby
customer = Customer.find(params[:customer_id])
order = Order.new(strong_params.merge(customer: customer))
```

Could brakeman be updated to know that ActiveRecord::Base#find and friends will make a parameter safe for mass assignment?

**Comment by presidentbeef** [comment]
Can you please share the exact warning message? Thanks.

**Comment by JasonBarnabe** [comment]
That was a simplified example, but here is a real one:

```ruby
  def create
    @affiliate_sales_order = AffiliateSalesOrder.find(params[:affiliate_sales_order_id])
    customer = @affiliate_sales_order.customer
    ti = TherapeuticInteraction.new(ti_params.merge(customer: customer, created_at: DateTime.now))
   end

  def ti_params
    params[:therapeutic_interaction_event].require(:therapeutic_interaction)
        .permit(:clinical_formulation_id)
  end
```

> Unprotected mass assignment near line 9: TherapeuticInteraction.new(ti_params.merge(:customer => AffiliateSalesOrder.find(params[:affiliate_sales_order_id]).customer, :created_at => DateTime.now))

**Comment by presidentbeef** [comment]
This is "fixed" by #1087.
